src/zenml/config/server_config.py

```diff
@@ -306,23 +318,25 @@ class ServerConfiguration(BaseModel):
 
     _deployment_id: Optional[UUID] = None
 
-    @root_validator(pre=True)
-    def _validate_config(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+    @model_validator(mode="before")
```

Comment:
Why not `after` validator?

Reply:
This is the case for both the old and the new validator here: It can take values that are not actually a part of the model. The validation here happens first and the processed result is passed to the actual pydantic validation.